daceml/autodiff/backward_pass_generator.py

```diff
@@ -0,0 +1,1270 @@
+                inputs.add(name)
+
+        outputs = set(backward_result.required_grad_names[name]
+                      for name in required_gradients)
+
+        for inp in inputs:
+            reverse_sdfg.arrays[inp].transient = False
+        for outp in outputs:
+            reverse_sdfg.arrays[outp].transient = False
+
+        # actually create the sdfg and return it
+        nsdfg = self.backward_state.add_nested_sdfg(
+            reverse_sdfg,
+            None,
+            inputs=inputs,
+            outputs=outputs,
+        )
+
+        for edge_args in deferred_edges:
+            edge_args["v"] = nsdfg
+            self.backward_state.add_edge(**edge_args)
+
+        return nsdfg, BackwardResult(
+            required_grad_names=backward_result.required_grad_names,
+            given_grad_names=backward_result.given_grad_names)
+
+    def _reverse_AccessNode(
+        self,
+        node: nd.AccessNode,
+        given_gradients: typing.List[str],
+        required_gradients: typing.List[str],
+    ) -> ReverseNodeReturnType:
+        rev = nd.AccessNode(self.array_grad_name(node.data),
+                            access=_invert_access(node.access))
+        self.backward_state.add_node(rev)
+        return rev, BackwardResult(required_grad_names={None: None},
+                                   given_grad_names={None: None})
+
+    def _reverse_MapEntry(
+        self,
+        node: nd.MapEntry,
+        given_gradients: typing.List[str],
+        required_gradients: typing.List[str],
+    ) -> ReverseNodeReturnType:
+
+        required_grad_names = {
+            n: _invert_map_connector(n)
+            for n in required_gradients
+        }
+        given_grad_names = {
+            n: _invert_map_connector(n)
+            for n in given_gradients
+        }
+        result = BackwardResult(required_grad_names=required_grad_names,
+                                given_grad_names=given_grad_names)
+        rev = nd.MapExit(self.reverse_map[node.map])
+
+        for conn in given_grad_names.values():
+            assert rev.add_in_connector(conn)
+
+        for conn in required_grad_names.values():
+            assert rev.add_out_connector(conn)
+
+        self.backward_state.add_node(rev)
+        return rev, result
+
+    def _reverse_MapExit(
+        self,
+        node: nd.MapExit,
+        given_gradients: typing.List[str],
+        required_gradients: typing.List[str],
+    ):
+        self.reverse_map[node.map] = copy.deepcopy(node.map)
+
+        rev = nd.MapEntry(self.reverse_map[node.map])
+        for conn in node.in_connectors:
+            assert rev.add_in_connector(conn)
+
+        for conn in node.out_connectors:
+            assert rev.add_out_connector(conn)
+
+        self.backward_state.add_node(rev)
+        return rev, BackwardResult(required_grad_names={
+            n: _invert_map_connector(n)
+            for n in required_gradients
+        },
+                                   given_grad_names={
```

Comment:
yapf sucks sometimes

Reply:
add parentheses around the return tuple please (`return (rev, ...)`). That should help. Otherwise, add a comma right after the last parameter as well